src/frontend/src/redux/middleware/request.ts

```diff
@@ -0,0 +1,112 @@
+import { Request } from '../../service/api-service';
+import axios from 'axios';
+import { Store, Action } from 'redux';
+
+// This Redux Action type shouldn't be used outside of here and service/api-service.ts
+export const REQUEST = 'REQUEST';
+
+// This Action and Middleware is a sort of implementation detail of the API Service needed
+// to allow API Requests to interact with the application state via the Redux store.
+export interface RequestAction {
+  type: typeof REQUEST;
+  request: Request;
+  resolve: Function;
+  reject: Function;
+}
+
+// This is a custom Middleware that does three things:
```

Comment:
This all looks well implemented, thanks for taking the time to do this! I feel like I would like to walk through it in person at some point and once it's merged I can see if I can figure anything out about the `any` types.